### DISCUSSION TITLE
Membuat coding dan boxplot dengan kondisi 2 selectInput

### DISCUSSION THREAD

**Discussion by angeliainge** [teamalgoritma/community#244] [Q&A] [answered]
Halo,

Mohon panduan untuk case sebagai berikut :

Saya sudah membuat coding untuk di Shiny Dashboard, dengan 2 selectinput yang depend on each other, jadi setelah "Country" dipilih, input "Job Title" akan mengikuti Job Title yang ada berdasarkan negara tersebut, contoh:

<img width="637" alt="Screen Shot 2022-08-03 at 6 57 20 AM" src="https://user-images.githubusercontent.com/107099548/182496297-c984309e-e2ce-4b56-b9e6-3e6f51a515d0.png">

Untuk dataframe nya : 
<img width="1007" alt="Screen Shot 2022-08-03 at 6 45 05 AM" src="https://user-images.githubusercontent.com/107099548/182496825-3e371c98-c9ba-4912-bdc7-ef5dd6c178e0.png">

<img width="1010" alt="Screen Shot 2022-08-03 at 6 45 15 AM" src="https://user-images.githubusercontent.com/107099548/182496404-62a483e9-0824-4bb7-975f-3e00dd510293.png">

yang saya inginkan adalah membuat kolom baru, dengan "average salary" merupakan rata2 salary berdasarkan job title yang sama dan hanya berlaku untuk negara yang sama (seperti di coret2an di atas). Namun saya tidak menemukan caranya, karena kalau group by (either by title / country), semua informasi nya di salah satu kolom tersebut akan di merge. 

Berikut coding yang saya coba buat:
<img width="833" alt="Screen Shot 2022-08-03 at 6 44 36 AM" src="https://user-images.githubusercontent.com/107099548/182496671-d658181d-eb08-4982-8652-5f7bc026a348.png">

Seperti di gambar, untuk "Country" (Austria), dan Job Title (Data Scientist) tergabung menjadi 1, dan kolom Experience level menghilang. Untuk semua average salary mempunyai value yang sama.

Goal saya adalah membuat boxplot dgn output saat di click country, memilih job title, lalu muncul 4 boxplot yang berisi average salary berdasarkan experience level (axis x)

<img width="477" alt="Screen Shot 2022-08-03 at 7 11 12 AM" src="https://user-images.githubusercontent.com/107099548/182497831-0580ba79-5a8e-4f5b-9659-34b56282ab3f.png">

Mohon panduan untuk penyelesaiannya dataframe beserta panduan untuk pembuatan boxplot dgn 2 input (saya sudah cari di google tapi tidak ketemu..)

**EDIT**
setelah saya pelajari kembali  mengenai boxplot (CMIIW), seharusnya tidak perlu membuat kolom baru yang menghitung average setiap job title berdasarkan negara dan experience level nya agar bisa mendapat nilai max and min nya juga?

Thank you

**Comment by inytss** [comment]
Dear @angeliainge ,
Betul sekali Bu, apabila ingin membuat sebuah boxplot, Ibu tidak perlu melakukan agregasi terhadap datanya, karena boxplot menggambarkan sebaran data. Apabila dilakukan agregasi, maka ketika Ibu melakukan pemilihan value kolom menggunakan `selectInput()` tidak akan terbentuk suatu boxplot melainkan hanya sebuah titik saja.

Saya izin untuk menandai pertanyaan Ibu sebagai "_Answered_" ya. Terima kasih :)

**Reply by angeliainge** [reply]
Halo kak, thank you replynya.
Sorry saya masih belum mendapat jawaban yang clear untuk step nya, untuk 2 kondisi selectinput ini apakah ada referensi coding yang bisa saya lihat?

Saya masih belum ada bayangan soalnya untuk step by step nya bagaimana

Yang saya pahami, axis X (job title) di pengaruhi oleh country yang dipilih, untuk coding ini saya harus menggunakn metode apa?

Thank you

**Comment by fafilia** [comment, marked as answer]
Halo @angeliainge, terkait case selectInput dari 2 variable yang berbeda dan saling berhubungan bisa Anda seleeaikan dengan mengatur di selectInput ke2 berdasarkan input yang dipilih pada selectInput yang pertama. Berikut link yang bisa saya berikan dimana casenya hampir mirip dengan case Anda. 

1. SelectInput dari 2 variable: https://stackoverflow.com/questions/27053493/r-shiny-using-reactive-renderui-value
2. Multiple input in shiny : https://stackoverflow.com/questions/67855470/dynamically-plot-multiple-variables-in-shiny-dashboard-r
3. R-Shiny two selectInput variables with partial dependency: https://stackoverflow.com/questions/38577474/r-shiny-two-selectinput-variables-with-partial-dependency
